fcrepo-http-api/src/test/java/org/fcrepo/http/api/services/HttpRdfServiceTest.java

```diff
@@ -254,42 +240,154 @@ public void testPatchToInternal_ExplicitSubjectObjectUpdate() {
+        final Predicate<Resource> keepDescId = a -> a.isURIResource() && a.hasURI(descriptionId.getFullId());
+        final var rdf = "@prefix dc: <"  + DC.getURI() + "> ." +
+                "@prefix dcterms: <"  + DCTerms.getURI() + "> ." +
+                "<" + binaryDescUri + "> dc:title 'fancy title' ;" +
+                "    dcterms:isPartOf <" + binaryUri + "> .";
+        final InputStream requestBodyStream = new ByteArrayInputStream(rdf.getBytes());
+        final Resource binaryRes = ResourceFactory.createResource(descriptionId.getFullDescribedId());
+        final var translated = httpRdfService.bodyToInternalModel(descriptionId, requestBodyStream, CONTENT_TYPE,
+                idTranslator, false);
+        assertTrue(translated.contains(binaryRes, DCTerms.isPartOf, binaryRes));
+        assertTrue(translated.contains(binaryRes, DC.title, ResourceFactory.createPlainLiteral("fancy title")));
+        assertFalse(translated.listSubjects().filterKeep(keepDescId).hasNext());
+    }
+
+    /**
+     * Test we do convert binary description URIs in the object of a triple.
+     */
+    @Test
+    public void testPutToInternal_BinaryDescriptionObject() {
+        final var binaryUri = HTTP_BASE_URI + UUID.randomUUID();
+        final var binaryDescUri = binaryUri + "/" + FCR_METADATA;
+        final var descriptionId = FedoraId.create(idTranslator.toInternalId(binaryDescUri));
+        final var rdf = "@prefix dc: <"  + DC.getURI() + "> ." +
+                "@prefix dcterms: <"  + DCTerms.getURI() + "> ." +
+                "<" + binaryUri + "> dc:title 'fancy title' ;" +
+                "    dcterms:isPartOf <" + binaryDescUri + "> .";
+        final InputStream requestBodyStream = new ByteArrayInputStream(rdf.getBytes());
+        final Resource binaryRes = ResourceFactory.createResource(descriptionId.getFullDescribedId());
+        final Resource binaryDescRes = ResourceFactory.createResource(descriptionId.getFullId());
+        final var translated = httpRdfService.bodyToInternalModel(descriptionId, requestBodyStream, CONTENT_TYPE,
+                idTranslator, false);
+        assertTrue(translated.contains(binaryRes, DCTerms.isPartOf, binaryRes));
+        assertTrue(translated.contains(binaryRes, DC.title, ResourceFactory.createPlainLiteral("fancy title")));
+        assertFalse(translated.listSubjects().filterKeep(a -> a.isURIResource() && a.hasURI(descriptionId.getFullId()))
+                .hasNext());
+        assertFalse(translated.listObjects()
+                .filterKeep(a -> a.isURIResource() && a.asResource().getURI().equals(descriptionId.getFullId()))
+                .hasNext());
+    }
+
+    /**
+     * Test that with an existing mismatch of binary and binary description IDs. We don't touch the old description
+     * triples anymore.
+     */
+    @Test
+    public void testMixedBinarySubjectsAlter() {
```

Comment:
So you want that I should remove the test then?

Reply:
I'm not sure. It's good to demonstrate the bug with a test, but it also doesn't seem like the bug will ever be fixed inside the fedora codebase (since the fix will be happening via another tool), and in theory the state shouldn't be being produced within Fedora itself anymore.